File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsJava/app/src/main/java/pl/udu/uwr/pum/roombasicsjava/view/WordListAdapter.java
package pl.udu.uwr.pum.roombasicsjava.view;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;

import pl.udu.uwr.pum.roombasicsjava.model.Word;

public class WordListAdapter extends ListAdapter<Word, WordViewHolder> {

    public WordListAdapter(@NonNull DiffUtil.ItemCallback<Word> diffCallback) {
        super(diffCallback);
    }

    @NonNull
    @Override
    public WordViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return WordViewHolder.create(parent);
    }

    @Override
    public void onBindViewHolder(WordViewHolder holder, int position) {
        Word current = getItem(position);
        holder.bind(current.getWord());
    }
}

File: 2022-2023/sourcecode/lab/lab9/ROOMbasicsKotlin/app/src/main/java/pl/udu/uwr/pum/roombasicskotlin/view/WordListAdapter.kt
package pl.udu.uwr.pum.roombasicskotlin.view

import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import pl.udu.uwr.pum.roombasicskotlin.model.Word

class WordListAdapter(wordsComparator: WordsComparator) : ListAdapter<Word, WordViewHolder>(wordsComparator) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WordViewHolder {
        return WordViewHolder.create(parent)
    }

    override fun onBindViewHolder(holder: WordViewHolder, position: Int) {
        val current = getItem(position)
        holder.bind(current.word)
    }
}